src/workerd/api/r2-bucket.h

```diff
@@ -256,15 +256,8 @@ class R2Bucket: public jsg::Object {
       return httpMetadata.map([](const HttpMetadata& m) { return m.clone(); });
     }
 
-    const jsg::Optional<jsg::Dict<kj::String>> getCustomMetadata() const {
-      return customMetadata.map([](const jsg::Dict<kj::String>& m) {
-        return jsg::Dict<kj::String>{
-          .fields =
-              KJ_MAP(f, m.fields) {
-          return jsg::Dict<kj::String>::Field{.name = kj::str(f.name), .value = kj::str(f.value)};
-        },
-        };
-      });
+    jsg::Optional<jsg::Dict<kj::String>> getCustomMetadata() {
+      return customMetadata.map([](jsg::Dict<kj::String>& m) { return kj::mv(m); });
```

Comment:
You seem to have changed this to move away the custom metadata, which means a second call to `getCustomMetadata()` would return an empty object. But since this is used to initialize a lazy property, it'll only be called once. If nothing else in the class uses `customMetadata` then maybe it's OK. If so, this needs some comments explaining.

But IMO that kind of change should not be hidden inside a commit which is supposed to be about changing the internal format of `jsg::Dict`. This change should probably be a separate PR, or at least a separate commit, to call it out, in case it breaks something.

Reply:
Definitely agree here. This should be called out or done separately. If this method is meant only to be ever called once to populate the lazy property, I would also consider adding a `KJ_ASSERT` before calling the `return customMetadata...`

Reply:
Reverted the change. I'll follow up in a different pull-request as suggested. Thank you for the reviews.